File: maestro-test/conformance-fixtures/flutter/lib/screens/tree_screen.dart
import 'package:flutter/material.dart';

/// contentDescriptor → a small tree with stable identifiers (identifier→resource-id,
/// label→content-desc). explicitChildNodes keeps the children as distinct nodes.
class TreeScreen extends StatelessWidget {
  const TreeScreen({super.key});

  @override
  Widget build(BuildContext context) {
    return Semantics(
      identifier: 'tree_root',
      label: 'tree_root',
      container: true,
      explicitChildNodes: true,
      child: Padding(
        padding: const EdgeInsets.symmetric(horizontal: 40, vertical: 80),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Semantics(
              identifier: 'tree_label_a',
              label: 'tree_label_a',
              child: const Padding(
                padding: EdgeInsets.symmetric(vertical: 20),
                child: Text('Label A', style: TextStyle(fontSize: 18)),
              ),
            ),
            Semantics(
              identifier: 'tree_button_b',
              label: 'tree_button_b',
              button: true,
              child: ElevatedButton(
                onPressed: () {},
                child: const Text('Button B'),
              ),
            ),
          ],
        ),
      ),
    );
  }
}

File: maestro-test/conformance-fixtures/compose/src/main/java/dev/mobile/maestro/fixture/screens/TreeScreen.kt
package dev.mobile.maestro.fixture.screens

import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Button
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.semantics.contentDescription
import androidx.compose.ui.semantics.semantics
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

@Composable
fun TreeScreen() {
    Column(
        modifier = Modifier
            .fillMaxSize()
            .padding(horizontal = 40.dp, vertical = 80.dp)
            .semantics { contentDescription = "tree_root" }
    ) {
        Text(
            text = "Label A",
            fontSize = 18.sp,
            modifier = Modifier
                .padding(vertical = 20.dp)
                .semantics { contentDescription = "tree_label_a" }
        )
        Button(
            onClick = { /* no event - tree oracle only */ },
            modifier = Modifier
                .semantics { contentDescription = "tree_button_b" }
        ) {
            Text("Button B")
        }
    }
}
